src/pybind/mgr/dashboard/services/service.py

```diff
@@ -54,42 +55,83 @@ def wait_for_refresh(orch, service_name, last_refreshed):
 
     while True:
         daemons = [d.to_dict() for d in orch.services.list_daemons(service_name=service_name)]
+        logger.info("Daemon list for service %s: %s", service_name, daemons)
+
+        if not daemons:
+            logger.info("No daemons found for service %s. Retrying...", service_name)
+            # Check if timeout has been reached
+            daemon_start_time = time.time()
+            if time.time() - daemon_start_time > timeout:
+                logger.error("Timeout reached while waiting for daemon list for service %s", service_name)
+                raise DashboardException(
+                    code='daemon_list_timeout',
+                    msg="Timeout reached while waiting for daemon list for service %s." % service_name
+                )
+            time.sleep(1)
+            continue  # Retry getting daemon list
+
         all_running = True
 
         for daemon in daemons:
+            logger.info("Checking daemon %s state", daemon['daemon_id'])
             daemon_state = daemon['status_desc']
-            logger.debug("Daemon %s state: %s", daemon['daemon_id'], daemon_state)
+            logger.info("Daemon %s state: %s", daemon['daemon_id'], daemon_state)
 
             if daemon_state in ('unknown', 'error', 'stopped'):
-                logger.error("Failed to restart daemon %s for service %s. State is %s", daemon['daemon_id'], service_name, daemon_state)  # noqa E501  # pylint: disable=line-too-long
+                logger.error("Failed to restart daemon %s for service %s. State is %s", daemon['daemon_id'], service_name, daemon_state)
                 raise DashboardException(
                     code='daemon_restart_failed',
-                    msg="Failed to restart the daemon %s. Daemon state is %s." % (service_name, daemon_state)  # noqa E501  # pylint: disable=line-too-long
+                    msg="Failed to restart the daemon %s. Daemon state is %s." % (daemon['daemon_id'], daemon_state)
                 )
-            if daemon_state != 'running':
+
+            if daemon_state == 'starting':
+                all_running = False
+
+            elif daemon_state != 'running':
                 all_running = False
 
         if all_running:
             logger.info("All daemons for service %s are running", service_name)
             return True
 
-        if time.time() - start_time > 10:
+        if time.time() - start_time > timeout:
             logger.error("Timeout reached while waiting for daemon %s to start", service_name)
             raise DashboardException(
                 code='daemon_restart_timeout',
                 msg="Timeout reached while waiting for daemon %s to start." % service_name
             )
-        return False
+
+        time.sleep(1)  # Adding a short delay before retrying
 
 
 class RgwServiceManager:
+    def create_rgw_instance(self, service_spec):
+        # Create RGW instance.
+        logger.info("Creating RGW instance")
```

Comment:
Be more specific. E.g.:
```suggestion
        logger.info("Creating RGW instance %s", service_spec.id)
```